readthedocs/core/resolver.py

```diff
@@ -135,7 +139,9 @@ def resolve_domain(project, private=None):
     relation = project.superprojects.first()
     subdomain = getattr(settings, 'USE_SUBDOMAIN', False)
     prod_domain = getattr(settings, 'PRODUCTION_DOMAIN')
-    public_domain = getattr(settings, 'PUBLIC_DOMAIN', prod_domain)
+    public_domain = getattr(settings, 'PUBLIC_DOMAIN', None)
+    if public_domain is None:
+        public_domain = prod_domain
```

Comment:
Oh, I get it now --  we're protecting against `PUBLIC_DOMAIN` being `None`, not allowing it. I misread the initial reply.